### PR TITLE
avoid skipping ways if mixing encoders in OSMRoadClassParser

### LINKED ISSUE #1843: Issue RoadClassEncoder via EncodedValueLookup when using mtb and motorway
When I set multiple vehicles in the config, including mtb, for example: 
```
graph.flag_encoders: car, mtb
```

My config looks like this:
```
graph.encoded_values: road_class,road_class_link,road_environment,max_speed,road_access
graph.bytes_for_flags: 8
```

If I lookup an encoded value in `InstructionsFromEdges` via `evLookup.getEnumEncodedValue(RoadClass.KEY, RoadClass.class);` I get an EncodedValue as expected.

For motorways this seems returns "other" instead of "motorway". Other road classes seem to be fine. If I only use car it works fine. 

I cross tested with `graph.flag_encoders: car, bike` and this particular issue did not exist. I don't think it's necessarily related to the EncodedValueLookup, but that's how I discovered this.

An example route where this happens is this (I think it happens for all motorways): http://localhost:8989/maps/?point=49.303926%2C8.627148&point=49.302688%2C8.626859

Maybe something is wrong with the number of bits reserved by the mtb encoder?

### PR DESCRIPTION
**Pull request by karussell**
fixes #1843

The reason why there is a difference for bike vs. mtb: in AcceptWay we use HashMap and getAccess uses the first entry which changes for different String keys. 

 - [x] remove missleading AcceptWay.getAccess

### CODE REVIEW

File: core/src/main/java/com/graphhopper/routing/util/parsers/OSMRoadClassParser.java [hunk 1]
```diff
@@ -43,8 +42,8 @@ public void createEncodedValues(EncodedValueLookup lookup, List<EncodedValue> li
     }
 
     @Override
-    public IntsRef handleWayTags(IntsRef edgeFlags, ReaderWay readerWay, EncodingManager.Access access, IntsRef relationFlags) {
-        if (!access.isWay())
+    public IntsRef handleWayTags(IntsRef edgeFlags, ReaderWay readerWay, boolean ferry, IntsRef relationFlags) {
+        if (ferry)
```
Comment (karussell) [inline core/src/main/java/com/graphhopper/routing/util/parsers/OSMRoadClassParser.java]: This is the actual fix.

Comment (boldtrn) [reply core/src/main/java/com/graphhopper/routing/util/parsers/OSMRoadClassParser.java]: Ah I see, that explains the issue :+1:.

Comment (boldtrn) [reply core/src/main/java/com/graphhopper/routing/util/parsers/OSMRoadClassParser.java]: BTW: do we need to pass the ferry boolean? We pass the that should contain the ferry information? Or is this for some ferry relations or something like this?

**Comment by karussell** [comment]
> We already pass the ReaderWay which should contain this information anyway

Yes, true. I would do this later when we remove the FlagEncoders completely. E.g. I'm unsure if we need the separate `encodingManager.acceptWay` method. Like you said: it might be slightly faster with it, and if it is not e.g. at least 10% faster it is certainly not necessary. And when we remove that method we could remove the ferry parameter too.

Another solution would be to create artificial tags like we already do with `estimated_distance` or set a ferry EncodedValue and ensure that it is available in every TagParser.

**Pull request merged**